Detectors/GRP/calibration/include/GRPCalibration/LHCIFfileReader.h

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef GRP_LHCIF_FILE_READER_H_
+#define GRP_LHCIF_FILE_READER_H_
+
+#include "Rtypes.h"
+#include <gsl/span>
+#include "Framework/Logger.h"
+#include <boost/tokenizer.hpp>
+#include <boost/lexical_cast.hpp>
+
+/// @brief Class to read the LHC InterFace file coming from the DCS filepush service
+
+namespace o2
+{
+namespace grp
+{
+class LHCIFfileReader
+{
+ public:
+  LHCIFfileReader() = default;  // default constructor
+  ~LHCIFfileReader() = default; // default destructor
+
+  void loadLHCIFfile(const char* fileName);            // load LHCIF file
+  void loadLHCIFfile(gsl::span<const char> configBuf); // load LHCIF file from buffer
+  template <typename T>
+  void readValue(const char* alias, std::string& type, int& nel, int& nmeas, std::vector<std::pair<float, std::vector<T>>>& meas);
+
+ private:
+  std::string mFileBuffStr; // buffer containing content of LHC IF file
+
+  ClassDefNV(LHCIFfileReader, 1);
+};
+
+template <typename T>
+void LHCIFfileReader::readValue(const char* alias, std::string& type, int& nele, int& nmeas, std::vector<std::pair<float, std::vector<T>>>& meas)
```

Comment:
better use `const std::string& alias`.
Why do you need `nmeas`, isn't `meas.second.size()` enough?

Reply:
Ok for std::string instead of char*. 
As for nmeas, I store it just because it is in the LHC-IF file, in case we want to use it as a x-check with what we get (for now I don't do it).